js-miniapp-sample/src/pages/user-details.js

```diff
@@ -149,20 +153,46 @@ export const dataFetchReducer = (state: State, action: Action) => {
         isLoading: true,
         isError: false,
         hasRequestedPermissions: false,
+        hasRequestedPointPermissions: false,
       };
     case 'FETCH_SUCCESS':
       return {
         ...state,
         isLoading: false,
         isError: false,
         hasRequestedPermissions: true,
+        hasRequestedPointPermissions: false,
       };
     case 'FETCH_FAILURE':
       return {
         ...initialState,
         isLoading: false,
         isError: true,
       };
+
+    case 'POINTS_FETCH_INIT':
+      return {
+        ...state,
+        isLoading: true,
+        isError: false,
+        hasRequestedPermissions: false,
+        hasRequestedPointPermissions: false,
+      };
```

Comment:
I think you should create specific properties for points for all of these properties. Otherwise there could be some weird issues with errors showing in the wrong place. Also then you don't need to set `isLoading`, `isError`, etc because `...state` will bring in those properties if they already exist. For example:
```suggestion
      return {
        ...state,
        isPointsLoading: true,
        isPointsError: false,
        hasRequestedPointPermissions: false,
      };
```